### PR TITLE
NO-SNOW Thread-safety check quote/backtick fix

### PR DESCRIPTION
**Pull request by sfc-gh-aalam**
1. Which Jira issue is this PR addressing? Make sure that there is an accompanying issue to your PR.

   

   Fixes SNOW-NNNNNNN

2. Fill out the following pre-review checklist:

   - [ ] I am adding a new automated test(s) to verify correctness of my new code
      - [ ] If this test skips Local Testing mode, I'm requesting review from @snowflakedb/local-testing
   - [ ] I am adding new logging messages
   - [ ] I am adding a new telemetry message
   - [ ] I am adding new credentials
   - [ ] I am adding a new dependency
   - [ ] If this is a new feature/behavior, I'm adding the Local Testing parity changes.
   - [x] I acknowledge that I have ensured my changes to be thread-safe. Follow the link for more information: [Thread-safe Developer Guidelines](https://docs.google.com/document/d/162d_i4zZ2AfcGRXojj0jByt8EUq-DrSHPPnTa4QvwbA/edit#bookmark=id.e82u4nekq80k)

3. Please describe how your code solves the related issue.

   let thread-safe check handle quotes, back-ticks when they are open and closed.

# TEST including backticks and quotes

"double quotes"
"open double quote
'single quotes'
'open single quote
`back ticks`
`open back tick

### CODE REVIEW

File: src/snowflake/snowpark/dataframe.py [hunk 1]
```diff
@@ -22,6 +22,7 @@
     overload,
 )
 
+#dummy change
```
Comment (sfc-gh-aalam) [inline src/snowflake/snowpark/dataframe.py:25]: added for testing. will remove before merging PR

Comment (sfc-gh-aalam) [reply src/snowflake/snowpark/dataframe.py:25]: this is also causing lint failure. removing the line would fix this issue.

**Review by sfc-gh-jrose — APPROVED** [review]

File: src/snowflake/snowpark/dataframe.py [hunk 2]
```diff
@@ -4335,3 +4335,4 @@ def print_schema(self) -> None:
     # joinTableFunction = join_table_function
     # naturalJoin = natural_join
     # withColumns = with_columns
+    # dummy comment
```
Comment (sfc-gh-mvashishtha) [inline src/snowflake/snowpark/dataframe.py:4338]: don't forget :)

**Review by sfc-gh-mvashishtha — APPROVED** [review]
LGTM @sfc-gh-aalam; thank you!

**Review by sfc-gh-helmeleegy — APPROVED** [review]
LGTM!

File: src/snowflake/snowpark/dataframe.py (hunk 2 above)
Comment (sfc-gh-aalam) [reply src/snowflake/snowpark/dataframe.py]: done :)

**Pull request merged**